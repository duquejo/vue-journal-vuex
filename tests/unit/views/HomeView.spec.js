import { shallowMount } from "@vue/test-utils";
import HomeView from '@/views/HomeView';

describe('HomeView.vue', () => {
  it('Should match the snapshot', () => {
    const wrapper = shallowMount(HomeView);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should redirect to no-entry view if any button is clicked', () => {
    const mockRouter = {
      push: jest.fn(),
    };

    const wrapper = shallowMount(HomeView, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });

    wrapper.find('button').trigger('click');

    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledTimes(1);
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });

  });
});