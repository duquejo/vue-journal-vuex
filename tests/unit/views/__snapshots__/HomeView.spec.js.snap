// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView.vue Should match the snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hello world</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</div>
`;
